packages/shell-api/src/database.ts

```diff
@@ -181,15 +234,15 @@ export default class Database extends ShellApiClass {
     this._emitDatabaseApiCall('aggregate', { options, pipeline });
 
     const {
-      providerOptions,
+      aggOptions,
       dbOptions,
       explain
     } = adaptAggregateOptions(options);
 
     const providerCursor = this._mongo._serviceProvider.aggregateDb(
       this._name,
       pipeline,
-      providerOptions,
+      { ...aggOptions, ...this._baseOptions },
```

Comment:
Maybe this is super nitpicky - but I think the order of arguments is swapped here and it should be `{ ...this._baseOptions, ...aggOptions}` 🤔 Other wise the `_baseOptions` would override anything in the `aggOptions`. Inside `collections.ts` the order is as basically everywhere else with `_baseOptions` first: https://github.com/mongodb-js/mongosh/pull/423/files#diff-c975be9c4e061c8d6cf6aac2dd4da42c6096bdf5e8e62601eca89f85e22302c3R130.

Reply:
Great catch! Not a nitpick at all :)